File: 21-merge-two-sorted-lists/merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode head = ListNode(0);
        ListNode *cur = &head;
        ListNode *c1 = list1;
        ListNode *c2 = list2;
        while(c1 != NULL && c2 != NULL) {
            if(c1->val < c2->val) {
                cur->next = c1;
                c1 = c1->next;
            } else {
                cur->next = c2;
                c2 = c2->next;
            }
            cur = cur->next;
        }
        if(c1 == NULL) {
            cur->next = c2;
        } else {
            cur->next = c1;
        }
        return head.next;
    }
};

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        cur1 = list1
        cur2 = list2
        res = ListNode()
        cur = res
        while cur1 and cur2:
            if cur1.val < cur2.val:
                new = ListNode(cur1.val)
                cur1 = cur1.next
            else:
                new = ListNode(cur2.val)
                cur2 = cur2.next
            cur.next = new
            cur = cur.next
        
        if cur1:
            cur.next = cur1
        if cur2:
            cur.next = cur2
        
        return res.next